packages/protocol/contracts/identity/Random.sol

```diff
@@ -93,20 +95,31 @@ contract Random is IRandom, Ownable, Initializable {
    */
   function addRandomness(uint256 blockNumber, bytes32 randomness) internal {
     history[blockNumber] = randomness;
-    if (historySize == 0) {
-      historyFirst = block.number;
```

Comment:
Was this correct? I changed this to be `blockNumber` now, after seeing some bad behavior in my tests.

Reply:
Interesting question. My understanding is that this should always equal to `block.number`? What was the bad behavior?

Reply:
Mine was a special case in testing when I was passing arbitrary blocknumbers in rather than using the real current `block.number`.
The part I think was "incorrect" here before was using both, i.e. `history[blockNumber]` and `historyFirst = block.number` I think both should use the same. Like you said, should never be a problem but was confusing to me